### PR TITLE
THU-457: Add MPL 2.0 License text to every source file

### PR DESCRIPTION
**Pull request by darkbanjo**
> [!NOTE]
> **Low Risk**
> Primarily mechanical header insertions plus a small CI/pre-commit enforcement script; low functional risk aside from potential edge cases in header insertion (shebangs/"use" directives/frontmatter).
> 
> **Overview**
> Adds the MPL 2.0 short-form license header to a large number of source files (JS/TS, tests, SQL, shell, HTML, Rust, etc.) to standardize licensing across the repo.
> 
> Introduces `scripts/license-headers.ts`, a Bun-based tool to *check or automatically insert* headers (handling shebangs, top-level "use" directives, and Astro frontmatter), and wires it into development/CI via `.lintstagedrc` and new `package.json` scripts (`license:check`, `license:fix`) included in `bun run check`.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit edfce2f0ec0c6038dd385858b1b3d3440dd580b5. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: scripts/license-headers.ts [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+ * Wired up in two places:
+ *   - .lintstagedrc → runs on staged source files in pre-commit hook
+ *   - package.json `license:check` → wired into `bun run check` for CI
+ */
+
+import { execSync } from 'node:child_process'
+import { readFileSync, writeFileSync } from 'node:fs'
+import { extname, resolve } from 'node:path'
+
+const REPO_ROOT = resolve(import.meta.dir, '..')
```
Comment (cjroth) [inline scripts/license-headers.ts]: Let's stick to camel case here

**Review by cjroth — CHANGES_REQUESTED** [review]
Other than the variable name convention thing, it looks good
